runtime/drivers/duckdb/transporter/objectStore_to_duckDB.go

```diff
@@ -260,3 +267,53 @@ func (a *appender) scanSchemaFromQuery(ctx context.Context, qry string) (map[str
 
 	return schema, nil
 }
+
+func (t *objectStoreToDuckDB) ingestDuckDBSQL(
+	ctx context.Context,
+	originalSQL string,
+	iterator drivers.FileIterator,
+	dbSink *drivers.DatabaseSink,
+	opts *drivers.TransferOpts,
+	p drivers.Progress,
+) error {
+	allFiles := make([]string, 0)
+	for iterator.HasNext() {
+		files, err := iterator.NextBatch(opts.IteratorBatch)
+		if err != nil {
+			return err
+		}
+		allFiles = append(allFiles, files...)
+	}
```

Comment:
I think the iterator deletes all the files from the previous iteration: https://github.com/rilldata/rill/blob/7e19861b2790db8d7ace6e6b4ebeaf1eb7f09ff8/runtime/drivers/blob/blobdownloader.go#L167C33-L167C33

Might need to add an iterator option to retain files until the iterator is closed.